rpc/src/transfer.rs

```diff
@@ -0,0 +1,202 @@
+use automata_primitives::{AccountId, Block, BlockId, Index};
+use automata_runtime::apis::TransferApi as TransferRuntimeApi;
+use fp_rpc::EthereumRuntimeRPCApi;
+use frame_system_rpc_runtime_api::AccountNonceApi;
+use jsonrpc_core::{Error, ErrorCode, Result};
+use jsonrpc_derive::rpc;
+use pallet_transfer::{eth_recover, evm_address_to_account_id_bytes};
+use sc_light::blockchain::BlockchainHeaderBackend as HeaderBackend;
+use sp_api::ProvideRuntimeApi;
+use sp_core::ecdsa;
+use sp_runtime::{codec::Decode, traits::Block as BlockT};
+use std::sync::Arc;
+
+const RUNTIME_ERROR: i64 = 1;
+
+#[rpc]
+pub trait TransferServer<BlockHash> {
+    //transfer to substrate address
+    #[rpc(name = "transfer_transferToSubstrateAccount")]
+    fn transfer_to_substrate_account(&self, message: String, signature: String) -> Result<u64>;
+
+    #[rpc(name = "transfer_transferNonce")]
+    fn transfer_nonce(&self, evm_addr: String) -> Result<u32>;
+}
+
+pub struct TransferApi<C> {
+    client: Arc<C>,
+}
+
+impl<C> TransferApi<C> {
+    pub fn new(client: Arc<C>) -> Self {
+        TransferApi { client }
+    }
+}
+
+impl<C> TransferServer<<Block as BlockT>::Hash> for TransferApi<C>
+where
+    C: Send + Sync + 'static,
+    C: ProvideRuntimeApi<Block> + HeaderBackend<Block>,
+    C::Api: TransferRuntimeApi<Block>
+        + EthereumRuntimeRPCApi<Block>
+        + AccountNonceApi<Block, AccountId, Index>,
+{
+    fn transfer_to_substrate_account(&self, message: String, signature: String) -> Result<u64> {
```

Comment:
same 0x handling and size check would be needed

Reply:
No need to check `0x` and String's
if `message` or `signature` has `0x` prefix, there will be `InvalidHexCharacter` error
if String's size is not legal, the bytes length check will fail